### PR TITLE
feat(sso): support saml integration

### PR DESCRIPTION
**Pull request by ungreat**
#### What type of PR is this?

Based on spring security，sso integration supports the saml protocol.

Here are same core class:
SamlParameter: Configuration class for SAML.
AddableRelyingPartyRegistrationRepository:  Responsible for loading configuration from the database
SamlCredentialManager：  Generate a pair of PrivateKey and X509Certificate, for saml's config.
CustomSamlProvider : A spring security provider to load or create a ODC user.

 
add generateSSOCredential api, to generate a pair of PrivateKey and X509Certificate.

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/starters/web-starter/src/main/java/com/oceanbase/odc/service/iam/auth/saml/CustomSamlProvider.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+public class CustomSamlProvider implements AuthenticationProvider {
+
+    private final AuthenticationProvider defaultAuthenticationProvider;
+
+    private final DefaultSamlUserService defaultSamlUserService;
+
+    public CustomSamlProvider(DefaultSamlUserService defaultSamlUserService) {
+        this.defaultSamlUserService = defaultSamlUserService;
+        defaultAuthenticationProvider = new OpenSamlAuthenticationProvider();
```
Comment (MarkPotato777) [inline server/starters/web-starter/src/main/java/com/oceanbase/odc/service/iam/auth/saml/CustomSamlProvider.java:42]: Why not just extend `OpenSamlAuthenticationProvider`?

File: server/starters/web-starter/src/main/java/com/oceanbase/odc/service/info/WebInfoAdapter.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ public String getLoginUrl(HttpServletRequest request) {
             return alipayOpenApiProperties.getObOfficialLoginUrl();
         } else if (authType.contains("local")) {
             SSOIntegrationConfig sSoClientRegistration = integrationService.getSSoIntegrationConfig();
-            if (sSoClientRegistration == null || !sSoClientRegistration.isOauth2OrOidc()) {
+            if (sSoClientRegistration == null) {
```
Comment (MarkPotato777) [inline server/starters/web-starter/src/main/java/com/oceanbase/odc/service/info/WebInfoAdapter.java:69]: What's this change for?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/saml/AddableRelyingPartyRegistrationRepository.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.oceanbase.odc.core.shared.Verify;
+import com.oceanbase.odc.service.integration.IntegrationService;
+import com.oceanbase.odc.service.integration.model.SSOIntegrationConfig;
+
+@Component
+public class AddableRelyingPartyRegistrationRepository implements RelyingPartyRegistrationRepository {
+
+    public final Cache<String, SSOIntegrationConfig> testConfigRegistrations =
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/integration/saml/AddableRelyingPartyRegistrationRepository.java:35]: Why do we need a cache here?

File: server/starters/web-starter/src/main/java/com/oceanbase/odc/service/iam/auth/saml/CustomSamlProvider.java (hunk 1 above)
Comment (ungreat) [reply server/starters/web-starter/src/main/java/com/oceanbase/odc/service/iam/auth/saml/CustomSamlProvider.java:42]: 1、 OpenSamlAuthenticationProvider is final class
2、in my opinion, combination is better than extends

File: server/starters/web-starter/src/main/java/com/oceanbase/odc/service/info/WebInfoAdapter.java (hunk 2 above)
Comment (ungreat) [reply server/starters/web-starter/src/main/java/com/oceanbase/odc/service/info/WebInfoAdapter.java:69]: Adapte for saml,  the type should not be distinguished  here,  but inside method.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/saml/AddableRelyingPartyRegistrationRepository.java (hunk 3 above)
Comment (ungreat) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/integration/saml/AddableRelyingPartyRegistrationRepository.java:35]: For test login, test login config not require persist.

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Pull request merged**